src/rs/lib.rs

```diff
@@ -1000,6 +1091,7 @@ impl Connection {
             unsafe {
                 Api::ffi_ref().ConnectionClose.unwrap()(self.handle);
             }
+            self.clear_ctx_if_present();
```

Comment:
This call is causing segfault on linux.
The problem seems to be that GetContext before ConnectionClose call returns the correct context ptr value, but after ConnectionClose call it returns a garbage ptr value. Ctx cleanup code tries to convert the garbage ptr value back to Rust Box and then drop the memory and hit segfault. This behaviour does not manifest on windows.
I ran valgrind and did not find mem corruption before this segfault is hit.
Is GetContext after ConnectionClose well defined? @nibanks

Reply:
Treat `ConnectionClose` like `delete` for the connection. You cannot call anything on it afterwards. The underlying platforms _migh_ manage the memory a bit differently, or have lingering reference counts internally for a bit, but you cannot depend on that.